command/lock.go

```diff
@@ -165,6 +170,10 @@ func (c *LockCommand) Run(args []string) int {
 	go func() {
 		if err := c.startChild(script, childDone, passStdin); err != nil {
 			c.Ui.Error(fmt.Sprintf("%s", err))
+
+			if c.childExitCode {
+				c.childReturnedError = true
```

Comment:
Hmm - looking at this now I think you do need a lock to be totally safe, but it's not the `childLock`. In the case where we send a `SIGKILL` we fire that and move on, so the `wait` down in `startChild` could be writing to this boolean value while the code below is reading it.

How about creating a buffered channel of size 1 and putting the `startChild` error result in there, or else closing the channel? The code below could block on that to wait for the child to die and get the result, or it could poll that channel to get any return code if available. Blocking is probably better, with a timeout that will raise an error.